sql/core/src/test/scala/org/apache/spark/sql/StringFunctionsSuite.scala

```diff
@@ -141,6 +141,19 @@ class StringFunctionsSuite extends QueryTest with SharedSQLContext {
     // scalastyle:on
   }
 
+  test("binary overlay function") {
+    // non ascii characters are not allowed in the code, so we disable the scalastyle here.
+    val df = Seq((Array[Byte](1, 2, 3, 4, 5, 6, 7, 8, 9))).toDF("a")
+    checkAnswer(df.select(overlay(
+      $"a", Array[Byte](-1), 6)), Row(Array[Byte](1, 2, 3, 4, 5, -1, 7, 8, 9)))
+    checkAnswer(df.select(overlay(
+      $"a", Array[Byte](-1, -1, -1, -1), 7)), Row(Array[Byte](1, 2, 3, 4, 5, 6, -1, -1, -1, -1)))
+    checkAnswer(df.select(overlay(
+      $"a", Array[Byte](-1, -1), 7, 0)), Row(Array[Byte](1, 2, 3, 4, 5, 6, -1, -1, 7, 8, 9)))
+    checkAnswer(df.select(overlay($"a", Array[Byte](-1, -1, -1, -1, -1), 2, 4)),
+      Row(Array[Byte](1, -1, -1, -1, -1, -1, 6, 7, 8, 9)))
+  }
```

Comment:
Can you add a mix case?: overlay(string, binary) and overlay(binary, string)

Reply:
Are we determined to support `overlay(string, binary)` and `overlay(binary, string)`?

Reply:
Ur, I forgot the discussion. ok, its ok to keep as it is. If necessary to support that, we can make a follow-up pr for that.

Reply:
OK. Thanks for the decision. @maropu